pkg/storage/backend-storage/backend-storage.go

```diff
@@ -268,9 +268,13 @@ func CurrentPVCName(vmi *corev1.VirtualMachineInstance) string {
 }
 
 func HasPersistentTPMDevice(vmiSpec *corev1.VirtualMachineInstanceSpec) bool {
-	return vmiSpec.Domain.Devices.TPM != nil &&
-		vmiSpec.Domain.Devices.TPM.Persistent != nil &&
-		*vmiSpec.Domain.Devices.TPM.Persistent
+	if vmiSpec.Domain.Devices.TPM == nil {
+		return false
+	}
+	if vmiSpec.Domain.Devices.TPM.Enabled != nil && !*vmiSpec.Domain.Devices.TPM.Enabled {
+		return false
+	}
```

Comment:
Two more if's later in the patch would benefit for an extra helper function of the above logic.
I'd guess HasEnabledTPMDevice() ? Some comments to help with the difference of Enabled != Persistent would help too.

Reply:
Looks good!. I'd just mention on the first comment the jira ticket
https://issues.redhat.com/browse/CNV-55729